vpr/src/device/rr_graph_view.cpp

```diff
@@ -0,0 +1,25 @@
+/***************************************
+ * Methods for Object RRGraphView
+ ***************************************/
+#include "rr_graph_view.h"
+
+/****************************
+ * Constructors
+ ****************************/
+RRGraphView::RRGraphView(const t_rr_graph_storage& node_storage,
+                         const RRSpatialLookup& node_lookup)
+    : node_storage_(node_storage)
+    , node_lookup_(node_lookup) {
+}
+
+/****************************
+ * Accessors
+ ****************************/
+t_rr_type RRGraphView::node_type(RRNodeId node) const {
+    return node_storage_.node_type(node);
+}
+
+const RRSpatialLookup& RRGraphView::node_lookup() const {
+    /* Return a constant object rather than a writable one */
```

Comment:
mmh, I am not a fan of comments stating the obvious. Problem with that is of course, that comments become noise instead of actual useful information, so readers of the code might learn to ignore comments even though it might contain something useful.

Also, the vertical space used is a lot, so that makes it harder to navigate the code (similar with the massive three-line comments about constructors and accessors which should't really be needed as anyone sees that anyway - but it sounds like this was decided for some other reason ? Still not a fan :) ).

Reply:
I think you are right. I have not written the comments carefully so that there are some redundant comments. Should remove this line. Also, I would like to share my opinions about writing comments (feel free to bug and share your experience)
- The comments about API's usage/functionality should be left in the header file, being close to the API declaration.
- The comments in source file should **NOT**  repeat what header files say. The comments in source file should focus on details of an API implementation, which is not easy to follow.

As such, developers can focus on reading header files for the key data structure and only look into specific functions in source files when necessary.

Reply:
Fully agree.